libraries/pipes/implementations/rust/src/types_ext.rs

```diff
@@ -26,11 +28,16 @@ impl Default for PipesContextData {
 }
 
 impl PipesMessage {
-    pub fn new(method: Method, params: Option<HashMap<String, Option<serde_json::Value>>>) -> Self {
+    pub fn new(method: Method, params: Option<HashMap<&str, Option<serde_json::Value>>>) -> Self {
         Self {
             dagster_pipes_version: "0.1".to_string(), // TODO: Make `const`
             method,
-            params,
+            params: params.map(|hashmap| {
+                hashmap
+                    .into_iter()
+                    .map(|(k, v)| (k.to_string(), v))
+                    .collect::<HashMap<String, Option<serde_json::Value>>>()
+            }),
```

Comment:
User's `.to_string` calls are delegated to the framework here. Better UX for a small iteration overhead.